### PR TITLE
ref(AlertRuleStatus): Split <AlertRuleStatus> into a component in its own file

### PR DESCRIPTION
**Pull request by ryan953**
I want to re-use this stuff, so we gotta pull it out from inside activatedRuleRow.tsx

### CODE REVIEW

File: static/app/views/alerts/list/rules/activatedRuleRow.tsx [hunk 1]
```diff
@@ -59,12 +53,9 @@ function ActivatedRuleListRow({
...
   const [assignee, setAssignee] = useState<string>('');
-  const isWaiting = useMemo(
-    () =>
-      !rule.activations?.length ||
-      (rule.activations?.length && rule.activations[0].isComplete),
-    [rule]
-  );
+  const isWaiting =
+    !rule.activations?.length ||
+    (rule.activations?.length && rule.activations[0].isComplete);
```
Comment (ryan953) [inline static/app/views/alerts/list/rules/activatedRuleRow.tsx:59]: just didn't need to be useMemo, it's a couple booleans!

File: static/app/views/alerts/list/rules/activatedRuleRow.tsx [hunk 2]
```diff
@@ -282,7 +210,9 @@ function ActivatedRuleListRow({
             />
           </Tooltip>
         </FlexCenter>
-        <MarginLeft>{renderAlertRuleStatus()}</MarginLeft>
+        <MarginLeft>
+          <AlertRuleStatus rule={rule} />
+        </MarginLeft>
```
Comment (ryan953) [inline static/app/views/alerts/list/rules/activatedRuleRow.tsx:215]: This is the 'Status' column from `/alerts/rules/`

File: static/app/views/alerts/list/rules/activatedRuleRow.tsx [hunk 3]
```diff
@@ -79,69 +70,6 @@ function ActivatedRuleListRow({
...
   }
 
-  function renderSnoozeStatus(): React.ReactNode {
-    return (
-      <IssueAlertStatusWrapper>
-        <IconMute size="sm" color="subText" />
-        {t('Muted')}
-      </IssueAlertStatusWrapper>
-    );
-  }
```
Comment (ryan953) [inline static/app/views/alerts/list/rules/activatedRuleRow.tsx:89]: This got inlined into the main function. that was the only callsite.

**Review by michellewzhang — APPROVED** [review]
great ref!

**Pull request merged**